fflogsapi/reports/fight.py

```diff
@@ -231,6 +234,11 @@ def _prepare_data_filters(self, filters: dict[str, Any]) -> tuple[str, dict[str,
         elif filters['endTime'] > fight_end:
             raise ValueError('Cannot retrieve fight events after the fight has ended!')
 
+        # if filterExpressions have been set rehydrayte them into an expression string
+        for filter, key in filters.items():
+            if filter == 'filterExpression':
+                filters['filterExpression'] = construct_filter_expression_string(key)
```

Comment:
```suggestion
        # if filterExpressions have been set as a dict, convert them into an expression string
        if 'filterExpression' in filters and type(filters['filterExpression']) is dict:
            filters['filterExpression'] = construct_filter_expression_string(key)
```

Don't want this to compromise the use of plain filter expressions